control-plane/controllers/configentries/registrations_controller.go

```diff
@@ -109,7 +121,85 @@ func (r *RegistrationsController) registerService(ctx context.Context, log logr.
 	return nil
 }
 
-func (r *RegistrationsController) deregisterService(ctx context.Context, log logr.Logger, client *capi.Client, registration *v1alpha1.Registration) error {
+func (r *RegistrationsController) updateTermGWACLRole(log logr.Logger, client *capi.Client, registration *v1alpha1.Registration) error {
+	roles, _, err := client.ACL().RoleList(nil)
+	if err != nil {
+		return err
+	}
+
+	var role *capi.ACLRole
+	for _, r := range roles {
+		if strings.HasSuffix(r.Name, "terminating-gateway-acl-role") {
```

Comment:
The Helm chart allows operators to create multiple terminating gateways. ([`terminatingGateways.gateways[].name`](https://developer.hashicorp.com/consul/docs/k8s/helm#v-terminatinggateways-gateways) ) Each of the TGWs will have a different role and policy name based on the name defined in the Helm config.

There needs to be some way to associate the service registration to a specific TGW so that only that TGW is granted access to front that service.

Reply:
yeah I'm going to be opening a follow up PR that handles the look ups/indexing similar to how we do it for resources on api-gateway, this right now is to get the base case of "one terminating gateway" working

Reply:
ended up being a smaller lift than I expected so just merged it in here